RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/ThreePIDModels.swift

```diff
@@ -0,0 +1,122 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+enum RegisterThreePID {
+    case email(String)
+    case msisdn(msisdn: String, countryCode: String)
+}
+
+struct ThreePIDCredentials: Codable {
+    var clientSecret: String?
+    
+    var identityServer: String?
+    
+    var sessionID: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case clientSecret = "client_secret"
+        case identityServer = "id_server"
+        case sessionID = "sid"
+    }
+}
+
+struct ThreePIDData {
+    let email: String
+    let msisdn: String
+    let country: String
+    let registrationResponse: RegistrationThreePIDTokenResponse
+    let registrationParameters: RegistrationParameters
+    
+    var threePID: RegisterThreePID {
+        email.isEmpty ? .msisdn(msisdn: msisdn, countryCode: country) : .email(email)
+    }
+}
+
+extension ThreePIDData {
+    init(threePID: RegisterThreePID,
```

Comment:
If `email` and `msisdn` are mutually exclusive, could we not keep this separation in `ThreePIDData` as well, rather than flattening `email`, `msisdn` and `country`, whilst leaving some with empty strings?

Reply:
Yeah, I agree with this looking at it again. I was expecting to learn more about how the ThreePID stuff worked when implementing other screens but thats very clearly a better choice to make right now 🙏

Reply:
Haha, and because of the initialisers and computed property it didn't actually need any changes anyway 🙃
That's much cleaner now.